pkg/inventory/inventory.go

```diff
@@ -192,19 +192,43 @@ func (i *Inventory) BuildVmCache(discoveredVmMap map[string]*runtimev1alpha1.Vir
 		}
 		if err != nil {
 			i.log.Error(err, "failed to update vm in vm cache", "vm", discoveredVm.Name,
-				"account", namespacedName.String())
+				"account", accountNamespacedName.String(), "selector", selectorNamespacedName.String())
 		}
 	}
 
 	if numVmsToAdd != 0 || numVmsToUpdate != 0 || numVmsToDelete != 0 {
-		i.log.Info("Vm poll statistics", "account", namespacedName, "added", numVmsToAdd,
-			"update", numVmsToUpdate, "delete", numVmsToDelete)
+		i.log.Info("Vm poll statistics", "account", accountNamespacedName, "selector", selectorNamespacedName,
+			"added", numVmsToAdd, "update", numVmsToUpdate, "delete", numVmsToDelete)
+	}
+}
+
+// DeleteAllVmsFromCache deletes all entries from vm cache for a given account.
+func (i *Inventory) DeleteAllVmsFromCache(accountNamespacedName *types.NamespacedName) error {
+	vmsInCache, err := i.vmStore.GetByIndex(indexer.VirtualMachineByAccountNamespacedName, accountNamespacedName.String())
+	if err != nil {
+		return err
 	}
+	var numVmsToDelete int
+	for _, cachedObject := range vmsInCache {
+		cachedVm := cachedObject.(*runtimev1alpha1.VirtualMachine)
+		key := fmt.Sprintf("%v/%v", cachedVm.Namespace, cachedVm.Name)
+		if err := i.vmStore.Delete(key); err != nil {
```

Comment:
Can we find a better way to delete objects. I mean there should be a better way to delete the object rather than fetching the key. Please evaluate